Lib/test/test_http/test_http.py

```diff
@@ -0,0 +1,11 @@
+import unittest
+import re
+
+class TestHTTP(unittest.TestCase):
+
+    def test_http_parse_request(self):
+        self.assertEqual(re.sub(r'^/+', '/', '//test.com'), '/test.com', '//test.com should be converted to a proper relative path')
+        self.assertEqual(re.sub(r'^/+', '/', '///test.com'), '/test.com', '///test.com should be converted to a proper relative path')
```

Comment:
This doesn't test anything useful. It only checks that some regular expression matches two strings. Since your code change is in `http.server`, your test should execute the `http.server` path that you changed.

A good test should fail without your code change to `http.server`, and pass with your change.

Reply:
Sorry I've been away due to my workload. But now as I've got some spare time I can do it better. So, as I've already said earlier that I'm new to writing these test cases and after checking out your comment I'm still confused about your requirements

Can you please elaborate these two statements:
> A good test should fail without your code change to http.server, and pass with your change.

and this one:
>A good test should fail without your code change to http.server, and pass with your change.

Reply:
@hamzaavvan, to put it in the simplest terms:
- on a clean checkout of the `main` branch, write a test that fails because it triggers the problem described in BPO-43223; this will prove we have a problem;
- *then* include your fix to prove that the fix solved the problem.

The test will have to actually run the HTTP server to be useful.